Lib/fontmake/font_project.py

```diff
@@ -438,9 +438,19 @@ def subset_otf_from_ufo(self, otf_path, ufo):
         ufo_order = [glyph_name
                      for glyph_name in ufo.lib[PUBLIC_PREFIX + 'glyphOrder']
                      if glyph_name in ufo]
-        for old_name, new_name in zip(
-                ufo_order,
-                TTFont(otf_path).getGlyphOrder()):
+        ot_order = TTFont(otf_path).getGlyphOrder()
+
+        # An OpenType binary usually has the .notdef glyph as the first glyph,
+        # which might or might not be present in the UFO, and not necessarily as
+        # the first glyph. Manipulate the order to make glyph names match up in
+        # the loop below.
+        if ".notdef" in ufo:
+            ufo_order.pop(ufo_order.index(".notdef"))
+            ufo_order.insert(0, ".notdef")
+        else:
+            ot_order = ot_order[1:]  # Strip out ".notdef"
```

Comment:
maybe you should check (or assert) that .notdef is really the first glyph in the TTFont. I mean, we know it is, because ufo2ft, but checking doesn't harm